proposals/subgroup-matrix.md

```diff
@@ -0,0 +1,715 @@
+*   Pointer: a logical pointer into an array (stride is ignored) (limited to Workgroup, StorageBuffer, and PhysicalStorageBuffer storage classes in Vulkan)
+*   Memory Layout: an enum for row- or column-majorness
+*   Stride: an optional operand indicating the stride between elements (aligned to min(16, align(col/row))).  The stride counts the number of elements in the pointee type; it is not a byte count.  For example, if the pointee type of the Pointer argument is 2xf16 then a stride of 8 translates to a byte stride of 8x (2x2) bytes.
+
+These operations convert to/from the abstract matrix type and memory. The operands **must** be uniform values for the scope.
+
+Additional operations:
+
+*   OpCooperativeMatrixLengthKHR: Number of accessible components from an invocation for a given cooperative matrix type.
+*   Conversions: All standard conversions are allowed (FToU, UToF, FToS, SToF, U, S, F, bitcast).
+*   Arithmetic: Negate, add, sub, mul, div using the opcode for the appropriate component type. Additionally, OpMatrixTimesScalar is also supported.
+*   Constants: OpCompositeConstruct and OpConstantComposite can provide a single fill value for a cooperative matrix.
+
+All operations involving a cooperative matrix **must** only be executed in uniform control flow for scope of the operation.
+
+Cooperative matrices **require** the Vulkan memory model in SPIR-V.
+
+OpTypeCooperativeMatrixKHR can only be instantiated by a variable in the Function or Private storage classes.
+
+
+### HLSL/Direct3D
+
+Microsoft originally proposed a feature, WaveMatrices, for
+[SM6.8](https://microsoft.github.io/DirectX-Specs/d3d/HLSL_SM_6_8_WaveMatrix.html)
+as experimental.
+This experimental feature was later withdrawn.
+
+Microsoft is now targeting a new feature, linalg::Matrix, for SM6.10.
+See https://github.com/microsoft/hlsl-specs/pull/556.
+The API side will likely look similar to the WaveMatrix feature, but it is not
+included in the proposal yet.
+
+The HLSL feature relies on SFINAE to provide a templated type for matrices.
+The type is templated with the following parameters:
+* ComponentType: Type of the matrix elements
+* M: Number of rows
+* N: Number of columns
+* Use: Use of the matrix: A, B, or Accumulator
+* Scope: The scope of the matrix: thread, wave, or threadgroup
+
+The level of functionality provided is a bit wider than that of SPV_KHR_cooperative_matrix.
+Additional functionality includes:
+* More type support: 16- and 64-bit integers, unorm and snorm formats.
+* Scope: Also supports workgroup scope matrices.
+* Conversions: A <-> B, A/B <-> Accumulator
+* Coordinates: Can access matrix coordinates when iterating over each thread's values.
+
+### MSL/Metal
+
+Apple calls them simdgroup matrices and support has existed since MSL 2.3. They are limited to relatively few GPUs though (Apple7+ in the [feature set](https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf)). That is iPhone 12+, some newer iPads, and the M1 and later macs.
```

Comment:
'relatively few GPUs' is potentially inaccurate, since Apple7 is A14 Bionic which released 5 years ago.

But I agree it is certainly not supported on all Metal devices. I.e., no Intel Mac support, many iPads do not support this, and some iPhones too.

Reply:
I'm capturing this feedback in an update to the proposal text in #5346 

There's no need to quantify it as many or few, just that it's not all of them.